### DISCUSSION TITLE
examples using Csla.Data.SqlClient

### DISCUSSION THREAD

**Discussion by chrisintampa** [MarimerLLC/csla#1720] [Questions]
All - I'm creating a new application and we have decided to upgrade to the new version of the csla utilizing MVC. we have been using  the .net framework version with webforms for years.

I'm running thru the project tracker app and I've gotten to the point where I am trying connecting to the database. it work fine with the EF, but I would like to avoid the use of the EF. I cant find an example using the Csla.Data.SqlClient.

I cant seem to figure out how to get a database connection - once I get a connection, i should be good to go.

any sample/instruction to  just doing a Fetch would be great.

**Comment by rockfordlhotka** [comment]
In modern .NET Core you probably shouldn't need things like the CSLA connection manager types. Instead you should rely on dependency injection of the connection into your data portal operation methods.

In `Startup` you can add a connection as a known type to the services collection so it can be injected. Typically you'll use `AddScoped` for this.

Then in your data portal operation method require the connect as an injected parameter.

```c#
  [Fetch]
  private async Task FetchAsync(int id, [Inject] SqlConnection sql)
  {
  }
```

This allows the .NET Core DI infrastructure to manage your connection.
